Framework/DataHandling/src/LoadILLIndirect.cpp

```diff
@@ -250,21 +249,20 @@ void LoadILLIndirect::loadDataIntoTheWorkSpace(
   Progress progress(this, 0, 1, m_numberOfTubes * m_numberOfPixelsPerTube +
                                     nb_monitors + nb_SD_detectors);
 
-  // Assign fake values to first X axis <<to be completed>>
-  for (size_t i = 0; i <= m_numberOfChannels; ++i) {
-    m_localWorkspace->dataX(0)[i] = double(i);
-  }
+  std::vector<double> increasingVals(m_numberOfChannels);
+  std::iota(increasingVals.begin(), increasingVals.end(), 1);
+  m_localWorkspace->mutableX(0) = increasingVals;
 
   // First, Monitor
   for (size_t im = 0; im < nb_monitors; im++) {
 
     if (im > 0) {
-      m_localWorkspace->dataX(im) = m_localWorkspace->readX(0);
+      m_localWorkspace->setSharedX(im, m_localWorkspace->sharedX(0));
     }
```

Comment:
You could actually remove the `if`, since `setSharedX` handles self-assignment correctly.